optuna/storages/_grpc/client.py

```diff
@@ -67,28 +73,35 @@ class GrpcStorageProxy(BaseStorage):
     """
 
     def __init__(self, *, host: str = "localhost", port: int = 13000) -> None:
-        self._stub = api_pb2_grpc.StorageServiceStub(
-            grpc.insecure_channel(
-                f"{host}:{port}",
-                options=[("grpc.max_receive_message_length", -1)],
-            )
-        )  # type: ignore
-        self._cache = GrpcClientCache(self._stub)
         self._host = host
         self._port = port
+        self.setup()
+
+    def setup(self) -> None:
+        self._channel = create_insecure_channel(self._host, self._port)
+        self._stub = api_pb2_grpc.StorageServiceStub(self._channel)
+        self._cache = GrpcClientCache(self._stub)
+
+    def wait_server_ready(self, timeout: float) -> None:
```

Comment:
It might be good to change the default value to None, which means wait forever until ready. This behavior is the same with `.result()` method.
https://grpc.github.io/grpc/python/grpc.html#grpc.Future.result

```suggestion
    def wait_server_ready(self, timeout: float | None = None) -> None:
```